Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/ApiGatewayHttpApiV2ProxyRequestTranslator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Amazon.Lambda.APIGatewayEvents;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using static Amazon.Lambda.APIGatewayEvents.APIGatewayHttpApiV2ProxyRequest;
+
+namespace Amazon.Lambda.TestTool
+{
+    /// <summary>
+    /// Translates ASP.NET Core HTTP requests to API Gateway HTTP API v2 Proxy Requests.
+    /// </summary>
+    public class ApiGatewayHttpApiV2ProxyRequestTranslator : IApiGatewayRequestTranslator
+    {
+        private readonly IHttpRequestUtility _httpRequestUtility;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiGatewayHttpApiV2ProxyRequestTranslator"/> class.
+        /// </summary>
+        /// <param name="httpRequestUtility">The HTTP request utility used for extracting request details.</param>
+        public ApiGatewayHttpApiV2ProxyRequestTranslator(IHttpRequestUtility httpRequestUtility)
+        {
+            _httpRequestUtility = httpRequestUtility;
+        }
+
+        /// <summary>
+        /// Translates an ASP.NET Core HTTP request to an API Gateway HTTP API v2 Proxy Request.
+        /// </summary>
+        /// <param name="request">The ASP.NET Core HTTP request to translate.</param>
+        /// <param name="pathParameters">The path parameters extracted from the request URL.</param>
+        /// <param name="resource">The API Gateway resource path.</param>
+        /// <returns>An <see cref="APIGatewayHttpApiV2ProxyRequest"/> object representing the translated request.</returns>
+        public object TranslateFromHttpRequest(HttpRequest request, IDictionary<string, string> pathParameters, string resource)
+        {
+            var (headers, _) = _httpRequestUtility.ExtractHeaders(request.Headers);
+            var (queryStringParameters, _) = _httpRequestUtility.ExtractQueryStringParameters(request.Query);
+
+            var httpApiV2ProxyRequest = new APIGatewayHttpApiV2ProxyRequest
+            {
+                RouteKey = $"{request.Method} {resource}",
+                RawPath = request.Path,
+                RawQueryString = request.QueryString.Value,
+                Cookies = request.Cookies.Select(c => $"{c.Key}={c.Value}").ToArray(),
```

Comment:
I believe you need to properly encode the cookies.

Reply:
https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html it wasnt clear from the docs to me what to do for those. do youhave any real example we can look at easily that you know of?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes looks like i should encode the cookie values to be safe

Reply:
ive added encoding the cookie value in 86b3f5ecbcbd1c0dc1d7f25db6ef1cfb313c12b4

Reply:
actually i just checked this again. 

when i logged the cookie in current production aws, i get

```
 headers: {
    'Content-Length': '0',
    Cookie: 'myCookie=special@value!++++',
```
when running with
```
$url = "https://myfunction.amazon.com/testfunction"
$cookieValue = "special@value!++++"

$headers = @{
    "Cookie" = "myCookie=$cookieValue"
}

$response = Invoke-WebRequest -Uri $url -Method GET -Headers $headers

# View the response status code
Write-Host "Status Code: $($response.StatusCode)"

# View the response content
Write-Host "Response Content:"
$response.Content

# Verify the HTTP version used
Write-Host "HTTP Version: $($response.Version)"

```

so it seems that we do not encode url encode it . so i am going to remove the cookie encoding then.